dotnet/Host.cs

```diff
@@ -0,0 +1,29 @@
+using System.Security.Cryptography;
+using Wasmtime;
+
+namespace WasmtimeDemo
+{
+    class Host : IHost
+    {
+        // These are from the current WASI proposal.
+        const int WASI_ENOTSUP = 58;
+        const int WASI_SUCCESS = 0;
+
+        public Instance Instance { get; set; }
+
+        [Import("fd_write", Module = "wasi_unstable")]
+        public int WriteFile(int fd, int iovs, int iovs_len, int nwritten)
+        {
+            return WASI_ENOTSUP;
+        }
+
+        [Import("random_get", Module = "wasi_unstable")]
```

Comment:
You should be able to switch "wasi_unstable" to "wasi_snapshot_preview1" here (and above), which shouldn't change anything for this demo, but it's nice to stay current.

Reply:
This would break the demo since currently it's importing from `wasi_unstable` with the latest `cargo-wasi` and `wasm32-wasi` target, afaik.

The C# implementation requires the import module name to match.

Reply:
This is another part of the demo I'm hoping will be ripped out once we can configure and use Wasmtime's WASI implementation from a C API.

Reply:
Ah, makes sense.